src/ImageBackendFactory.php

```diff
@@ -37,9 +37,15 @@ public function __construct(Factory $creator)
      */
     public function create($service, array $params = [])
     {
-        /** @var AssetContainer $assetContainer */
+        /** @var AssetContainer|null $assetContainer */
         $assetContainer = reset($params);
-        if (!$assetContainer instanceof AssetContainer) {
+
+        // If no asset container was passed in, create a new uncached image backend
+        if (!$assetContainer) {
+            return $this->creator->create($service, $params);
+        }
+
+        if (!($assetContainer instanceof AssetContainer)) {
```

Comment:
This used to be `if (!$assetContainer instanceof AssetContainer)` - which actually checks `if ((!$assetContainer) instanceof AssetContainer)`. In other words it was checking if a boolean was an instance of a class, which will always evaluate to false.